packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -29,24 +29,47 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
   ///
   /// The [padding] argument must not be null and must have non-negative insets.
   RenderSliverPadding({
-    @required EdgeInsets padding,
+    @required EdgeInsetsGeometry padding,
+    TextDirection textDirection,
     RenderSliver child,
   }) : assert(padding != null),
-       assert(padding.isNonNegative),
```

Comment:
I believe isNonNegative is actually on EdgeInsetsGeometry now. I may have failed to update Padding when I changed that.

Reply:
Looks like padding has it.  Not sure why I removed this.